src/terrain/routes/filesystem/tickets.clj

```diff
@@ -0,0 +1,58 @@
+(ns terrain.routes.filesystem.tickets
+  (:use [common-swagger-api.schema]
+        [ring.util.http-response :only [ok]]
+        [terrain.routes.schemas.filesystem.tickets]
+        [terrain.util :only [optional-routes]]
+        [terrain.util.transformers :only [add-current-user-to-map]])
+  (:require [common-swagger-api.schema.data :as data-schema]
+            [common-swagger-api.schema.data.tickets :as schema]
+            [terrain.clients.data-info :as data]
+            [terrain.util.config :as config]))
+
+;; FIXME: The `coerce-public-param` middleware is only for backwards compatibility
+;; and should be removed in the next version of the API
+(defn- public-param->boolean
+  "Sets the `public` key in `params` to `true` if its string value is `1` or `true`,
+   otherwise sets it to `false`."
+  [{public "public" :as params}]
+  (assoc params "public" (or (= public "1") (= public "true"))))
+
+(defn- coerce-public-param
+  "Middleware to convert public param string values of `1` to a boolean `true`."
+  [handler]
+  (fn [request]
+    (handler (update request :query-params public-param->boolean))))
+
+(defn filesystem-ticket-routes
+  "The routes for filesystem ticket endpoints."
+  []
+
+  (optional-routes
+    [config/filesystem-routes-enabled]
+
+    (context
+      "/filesystem" []
+      :tags ["filesystem"]
+
+      (POST "/tickets" []
+            :middleware [coerce-public-param]
```

Comment:
The `public` param used to be optional, defaulting to a `false` value passed along to the required `data-info` param. It also used to only pass `true` to `data-info` if terrain's param was set to `1`.

This middleware is for backwards compatibility, and allows either `1` or `true` to set the `data-info` param to `true`, and anything else will pass along a `false`. The Swagger docs on this endpoint will only show `true` or `false` options, though.